chainer/links/model/vision/resnet.py

```diff
@@ -0,0 +1,447 @@
+    and permute the dimensions before calling.
+
+    Args:
+        image (PIL.Image or numpy.ndarray): Input image.
+            If an input is ``numpy.ndarray``, its shape must be
+            ``(height, width)``, ``(height, width, channels)``,
+            or ``(channels, height, width)``.
+        size (pair of ints): Size of converted images.
+            If ``None``, the given image is not resized.
+
+    Returns:
+        numpy.ndarray: The converted output array.
+
+    """
+
+    if not available:
+        raise ImportError('PIL cannot be loaded. Install Pillow!\n'
+                          'The actual import error is as follows:\n' +
+                          str(_import_error))
+    if isinstance(image, numpy.ndarray):
+        if image.ndim == 3:
+            if image.shape[0] == 1:
+                image = image[0, :, :]
+            elif image.shape[0] == 3:
+                image = image.transpose((1, 2, 0))
+        image = Image.fromarray(image.astype(numpy.uint8))
+    image = image.convert('RGB')
+    if size is not None:
+        image = image.resize(size)
+    image = numpy.asarray(image, dtype=numpy.float32)
+    image = image[:, :, ::-1]
+    # NOTE: in the original paper they subtract a fixed mean image,
+    #       however, in order to support arbitrary size we instead use the
+    #       mean pixel (rather than mean image) as with VGG team. The mean
+    #       value used in ResNet is slightly different from that of VGG16.
+    image -= numpy.array(
+        [103.063,  115.903,  123.152], dtype=numpy.float32)
+    image = image.transpose((2, 0, 1))
+    return image
+
+
+class BuildingBlock(link.Chain):
+
+    """A building block that consists of several Bottleneck layers.
+
+    Args:
+        n_layer (int): Number of layers used in the building block.
+        in_channels (int): Number of channels of input arrays.
+        mid_channels (int): Number of channels of intermediate arrays.
+        out_channels (int): Number of channels of output arrays.
+        stride (int or tuple of ints): Stride of filter application.
+        initialW (4-D array): Initial weight value used in
+            the convolutional layers.
+    """
+
+    def __init__(self, n_layer, in_channels, mid_channels,
+                 out_channels, stride, initialW=None):
+        links = [
+            ('a', BottleneckA(
+                in_channels, mid_channels, out_channels, stride, initialW))
+        ]
+        for i in range(n_layer - 1):
+            name = 'b{}'.format(i + 1)
+            link = BottleneckB(out_channels, mid_channels, initialW)
+            links.append((name, link))
+        super(BuildingBlock, self).__init__(**dict(links))
+        self.forward = links
+
+    def __call__(self, x, test=True):
+        for name, func in self.forward:
+            x = func(x, test=test)
+        return x
+
+
+class BottleneckA(link.Chain):
+
+    """A bottleneck layer that reduces the resolution of the feature map.
```

Comment:
No. In this case in_channels < out_channels, and "reduces the resolution of the feature map" means the spatial resolution.